website/docs/plugin/framework/handling-data/types/dynamic.mdx

```diff
@@ -0,0 +1,146 @@
+
+</Tip>
+
+Access `types.Dynamic` information via the following methods:
+
+* [`(types.Dynamic).IsNull() bool`](https://pkg.go.dev/github.com/hashicorp/terraform-plugin-framework/types/basetypes#DynamicValue.IsNull): Returns true if the dynamic value is null.
+* [`(types.Dynamic).IsUnknown() bool`](https://pkg.go.dev/github.com/hashicorp/terraform-plugin-framework/types/basetypes#DynamicValue.IsUnknown): Returns true if the dynamic value is unknown.
+* [`(types.Dynamic).UnderlyingValue() attr.Value`](https://pkg.go.dev/github.com/hashicorp/terraform-plugin-framework/types/basetypes#DynamicValue.UnderlyingValue): Returns the underlying value of the dynamic container, will be `nil` if null or unknown.
+
+In this example, a dynamic value is checked for being null or unknown value first, before accessing its known value:
+
+```go
+// Example data model definition
+// type ExampleModel struct {
+//   ExampleAttribute types.Dynamic `tfsdk:"example_attribute"`
+// }
+//
+// This would be filled in, such as calling: req.Plan.Get(ctx, &data)
+var data ExampleModel
+
+// optional logic for handling null value
+if data.ExampleAttribute.IsNull() {
+    // ...
+}
+
+// optional logic for handling unknown value
+if data.ExampleAttribute.IsUnknown() {
+    // ...
+}
+
+// myDynamicVal now contains the underlying value, determined by Terraform at runtime
+myDynamicVal := data.ExampleAttribute.UnderlyingValue()
+```
+
+### Handling the Underlying Value
+
+If a dynamic value is known, a [Go type switch](https://go.dev/tour/methods/16) can be used to access the type-specific methods for data handling:
+
+```go
+	switch value := data.ExampleAttribute.UnderlyingValue().(type) {
+	case types.Bool:
+		// Handle boolean value
+	case types.Number:
+		// Handle float64, int64, and number values
+	case types.List:
+		// Handle list value
+	case types.Map:
+		// Handle map value
+	case types.Object:
+		// Handle object value
+	case types.Set:
+		// Handle set value
+	case types.String:
+		// Handle string value
+	case types.Tuple:
+		// Handle tuple value
+	}
+```
+
+The type of the underlying value is determined at runtime by Terraform if the value is from configuration. Developers dealing with dynamic data will need to have extensive knowledge of the [Terraform type system](/terraform/language/expressions/types) to properly handle all potential practitioner configuration scenarios.
+
+Refer to the [Dynamic Data Handling](/terraform/plugin/framework/handling-data/terraform-concepts#dynamic-data-handling-caveats) documentation for more information.
+
+## Setting Values
+
+Call one of the following to create a `types.Dynamic` value:
+
+* [`types.DynamicNull()`](https://pkg.go.dev/github.com/hashicorp/terraform-plugin-framework/types#DynamicNull): A null dynamic value.
+* [`types.DynamicUnknown()`](https://pkg.go.dev/github.com/hashicorp/terraform-plugin-framework/types#DynamicUnknown): An unknown dynamic value.
```

Comment:
Not sure if this is worth distinguishing?

```suggestion
* [`types.DynamicNull()`](https://pkg.go.dev/github.com/hashicorp/terraform-plugin-framework/types#DynamicNull): A null dynamic value.
* [`types.DynamicUnknown()`](https://pkg.go.dev/github.com/hashicorp/terraform-plugin-framework/types#DynamicUnknown): An unknown dynamic value where the final static type is not known. Use `types.DynamicValue()` with an unknown value if the final static type is known.
```

Reply:
Good call ⚡